src/exchanges/oasis/Eth2DaiDirect.js

```diff
@@ -0,0 +1,203 @@
+import PrivateService from '../../core/PrivateService';
+import { getCurrency, WETH, DAI, ETH } from '../../eth/Currency';
+import contracts from '../../../contracts/contracts';
+import { OasisSellOrder, OasisBuyOrder } from './OasisOrder';
+
+export default class Eth2DaiDirect extends PrivateService {
+  constructor(name = 'exchange') {
+    super(name, [
+      'proxy',
+      'smartContract',
+      'token',
+      'cdp',
+      'web3',
+      'transactionManager',
+      'allowance'
+    ]);
+    this._slippage = 0.02;
+  }
+
+  async sell(sell, buy, amount) {
+    const proxy = await this._checkProxy(sell);
+    const method = this._setMethod(sell, buy, 'sellAllAmount', proxy);
+    const sellToken = sell === 'ETH' ? 'WETH' : sell;
+    const buyToken = buy === 'ETH' ? 'WETH' : buy;
+    const minFillAmount = await this._minBuyAmount(buyToken, sellToken, amount);
+    const params = await this._buildParams(
+      sellToken,
+      amount,
+      buyToken,
+      minFillAmount,
+      method
+    );
+    const options = this._buildOptions(amount, sell, method);
+
+    if (proxy) await this.get('allowance').requireAllowance(sellToken, proxy);
+    return OasisSellOrder.build(
+      this._oasisDirect(),
+      method,
+      params,
+      this.get('transactionManager'),
+      buyToken === 'DAI' ? DAI : WETH,
+      options
+    );
+  }
+
+  async buy(buy, sell, amount) {
+    const proxy = await this._checkProxy(sell);
+    const method = this._setMethod(sell, buy, 'buyAllAmount', proxy);
+    const buyToken = buy === 'ETH' ? 'WETH' : buy;
+    const sellToken = sell === 'ETH' ? 'WETH' : sell;
+    const maxPayAmount = await this._maxPayAmount(sellToken, buyToken, amount);
+    const params = await this._buildParams(
+      buyToken,
+      amount,
+      sellToken,
+      maxPayAmount,
+      method
+    );
+    const options = this._buildOptions(amount, sell, method, maxPayAmount);
+
+    if (proxy) await this.get('allowance').requireAllowance(sellToken, proxy);
+    return OasisBuyOrder.build(
+      this._oasisDirect(),
+      method,
+      params,
+      this.get('transactionManager'),
+      options
+    );
+  }
+
+  setSlippageLimit(limit) {
+    return (this._slippage = limit);
```

Comment:
it's probably fine here just to do the assignment and return nothing -- seeing return followed by an assignment always looks a little shady to me, probably because it's easily mistaken for a boolean, e.g. `return x == y`